src/Sql/Billing/Tables/ProviderPlan.sql

```diff
@@ -0,0 +1,10 @@
+CREATE TABLE [dbo].[ProviderPlan] (
+    [Id]             UNIQUEIDENTIFIER NOT NULL,
+    [ProviderId]     UNIQUEIDENTIFIER NOT NULL,
+    [PlanType]       TINYINT          NOT NULL,
+    [SeatMinimum]    INT              NULL,
+    [PurchasedSeats] INT              NULL,
+    [AllocatedSeats] INT              NULL,
+    CONSTRAINT [PK_ProviderPlan] PRIMARY KEY CLUSTERED ([Id] ASC),
+    CONSTRAINT [FK_ProviderPlan_Provider] FOREIGN KEY ([ProviderId]) REFERENCES [dbo].[Provider] ([Id]) ON DELETE CASCADE
```

Comment:
❓ What's the intent with the cascade? My experience has taught me these are dangerous and difficult to build a smart logic layer on top of.

Reply:
IMO it doesn't make sense to have a floating `ProviderPlan` record that exists after the `Provider` it's related to has been deleted. It can't be used and serves no functional purpose, so I figured cascading the deletion off the `Provider` record would help us keep the table clean.

Reply:
Well, I took a look and we do this quite a bit. Not my preference but so be it, tough to change the position now.